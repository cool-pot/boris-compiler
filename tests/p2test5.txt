*** This is a sample file that should work OK with the PARSER.
*** but should cause terrible trouble once symbol table and type and scope checking are incorporated.

*** In short, this is a test input for project2/parser ONLY

array a [234..123] a = a*a*a - 2*a + 4;
global a=1;
local a=a;

a = a.0, a.0, 0;

foreach a in a do
  print a, a[a+1], a[a.7+49+a[a.3+5]];
end for

if a[0] < a.0 / a + 5 then
  print 999;
else
  print 777;
end if

defun a (a,aa,aaa,a) 
  local a=aa;
  global aaa=aa;

  if a[aa[aaa]]==aaa[a.7] then
     return aaa[a.5];
  else
     a[a[7]] <-> a.3;
     return a[a[a.3]];
  end if
end defun

while a != a.7 do
  a <-> a,a;
end while