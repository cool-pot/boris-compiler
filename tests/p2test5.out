> Hello from parserTester for boris compiler!
Start parse tree visualization:

[ROOT_INPUT]
    [SDD_LIST]
        [DECL_AS_SDD]
            [ARRAY_DECL_WITH_ANONY_FUNC]
                [PLACEHOLDER-258]array
                [ID]a
                [PLACEHOLDER-288][
                [SINGLE_INT_AS_EXPR]
                    [INT]234
                [PLACEHOLDER-275]..
                [SINGLE_INT_AS_EXPR]
                    [INT]123
                [PLACEHOLDER-289]]
                [ID]a
                [PLACEHOLDER-293]=
                [EXPR_MINUS_EXPR]
                    [EXPR_MULT_EXPR]
                        [EXPR_MULT_EXPR]
                            [SINGLE_ID_AS_EXPR]
                                [ID]a
                            [PLACEHOLDER-286]*
                            [SINGLE_ID_AS_EXPR]
                                [ID]a
                        [PLACEHOLDER-286]*
                        [SINGLE_ID_AS_EXPR]
                            [ID]a
                    [PLACEHOLDER-285]-
                    [EXPR_PLUS_EXPR]
                        [EXPR_MULT_EXPR]
                            [SINGLE_INT_AS_EXPR]
                                [INT]2
                            [PLACEHOLDER-286]*
                            [SINGLE_ID_AS_EXPR]
                                [ID]a
                        [PLACEHOLDER-284]+
                        [SINGLE_INT_AS_EXPR]
                            [INT]4
                [PLACEHOLDER-290];
        [SDD_LIST]
            [DECL_AS_SDD]
                [GLOBAL_DECL]
                    [PLACEHOLDER-261]global
                    [ID]a
                    [PLACEHOLDER-293]=
                    [SINGLE_INT_AS_EXPR]
                        [INT]1
                    [PLACEHOLDER-290];
            [SDD_LIST]
                [DECL_AS_SDD]
                    [LOCAL_DECL]
                        [PLACEHOLDER-260]local
                        [ID]a
                        [PLACEHOLDER-293]=
                        [SINGLE_ID_AS_EXPR]
                            [ID]a
                        [PLACEHOLDER-290];
                [SDD_LIST]
                    [STATEMENT_AS_SDD]
                        [LHS_ASSIGN_EXPR_AS_STATEMENT]
                            [LHS]
                                [SINGLE_ID_AS_LHSITEM]
                                    [ID]a
                                [EMPTY_NODE]
                            [PLACEHOLDER-293]=
                            [EXPR_COMMA_EXPR]
                                [EXPR_COMMA_EXPR]
                                    [TUPLE_REF_AS_EXPR]
                                        [ID]a
                                        [PLACEHOLDER-277].
                                        [INT]0
                                    [PLACEHOLDER-276],
                                    [TUPLE_REF_AS_EXPR]
                                        [ID]a
                                        [PLACEHOLDER-277].
                                        [INT]0
                                [PLACEHOLDER-276],
                                [SINGLE_INT_AS_EXPR]
                                    [INT]0
                            [PLACEHOLDER-290];
                    [SDD_LIST]
                        [STATEMENT_AS_SDD]
                            [FOREACH_ARRAYID_STATEMENT]
                                [PLACEHOLDER-270]foreach
                                [ID]a
                                [PLACEHOLDER-272]in
                                [ARRAY_ID]
                                    [ID]a
                                [PLACEHOLDER-265]do
                                [STATEMENT_LIST]
                                    [PRINT_STATEMENT]
                                        [PLACEHOLDER-274]print
                                        [EXPR_COMMA_EXPR]
                                            [EXPR_COMMA_EXPR]
                                                [SINGLE_ID_AS_EXPR]
                                                    [ID]a
                                                [PLACEHOLDER-276],
                                                [ARRAY_REF_AS_EXPR]
                                                    [ID]a
                                                    [PLACEHOLDER-288][
                                                    [EXPR_PLUS_EXPR]
                                                        [SINGLE_ID_AS_EXPR]
                                                            [ID]a
                                                        [PLACEHOLDER-284]+
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]1
                                                    [PLACEHOLDER-289]]
                                            [PLACEHOLDER-276],
                                            [ARRAY_REF_AS_EXPR]
                                                [ID]a
                                                [PLACEHOLDER-288][
                                                [EXPR_PLUS_EXPR]
                                                    [EXPR_PLUS_EXPR]
                                                        [TUPLE_REF_AS_EXPR]
                                                            [ID]a
                                                            [PLACEHOLDER-277].
                                                            [INT]7
                                                        [PLACEHOLDER-284]+
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]49
                                                    [PLACEHOLDER-284]+
                                                    [ARRAY_REF_AS_EXPR]
                                                        [ID]a
                                                        [PLACEHOLDER-288][
                                                        [EXPR_PLUS_EXPR]
                                                            [TUPLE_REF_AS_EXPR]
                                                                [ID]a
                                                                [PLACEHOLDER-277].
                                                                [INT]3
                                                            [PLACEHOLDER-284]+
                                                            [SINGLE_INT_AS_EXPR]
                                                                [INT]5
                                                        [PLACEHOLDER-289]]
                                                [PLACEHOLDER-289]]
                                        [PLACEHOLDER-290];
                                    [EMPTY_NODE]
                                [PLACEHOLDER-263]end
                                [PLACEHOLDER-271]for
                        [SDD_LIST]
                            [STATEMENT_AS_SDD]
                                [IF_ELSE_STATEMENT]
                                    [PLACEHOLDER-266]if
                                    [BOOLEXPR]
                                        [ARRAY_REF_AS_EXPR]
                                            [ID]a
                                            [PLACEHOLDER-288][
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]0
                                            [PLACEHOLDER-289]]
                                        [PLACEHOLDER-278]<
                                        [EXPR_PLUS_EXPR]
                                            [EXPR_DIV_EXPR]
                                                [TUPLE_REF_AS_EXPR]
                                                    [ID]a
                                                    [PLACEHOLDER-277].
                                                    [INT]0
                                                [PLACEHOLDER-287]/
                                                [SINGLE_ID_AS_EXPR]
                                                    [ID]a
                                            [PLACEHOLDER-284]+
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]5
                                    [PLACEHOLDER-267]then
                                    [STATEMENT_LIST]
                                        [PRINT_STATEMENT]
                                            [PLACEHOLDER-274]print
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]999
                                            [PLACEHOLDER-290];
                                        [EMPTY_NODE]
                                    [EMPTY_NODE]
                                    [ELSE_SENTENCE]
                                        [PLACEHOLDER-269]else
                                        [STATEMENT_LIST]
                                            [PRINT_STATEMENT]
                                                [PLACEHOLDER-274]print
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]777
                                                [PLACEHOLDER-290];
                                            [EMPTY_NODE]
                                    [PLACEHOLDER-263]end
                                    [PLACEHOLDER-266]if
                            [SDD_LIST]
                                [DEFN_AS_SDD]
                                    [FUNC_DEFN]
                                        [PLACEHOLDER-262]defun
                                        [ID]a
                                        [PLACEHOLDER-291](
                                        [ID]a
                                        [COMMA_ID_LIST]
                                            [PLACEHOLDER-276],
                                            [ID]aa
                                            [COMMA_ID_LIST]
                                                [PLACEHOLDER-276],
                                                [ID]aaa
                                                [COMMA_ID_LIST]
                                                    [PLACEHOLDER-276],
                                                    [ID]a
                                                    [EMPTY_NODE]
                                        [PLACEHOLDER-291](
                                        [BODY]
                                            [SD_LIST]
                                                [DECL_AS_SD]
                                                    [LOCAL_DECL]
                                                        [PLACEHOLDER-260]local
                                                        [ID]a
                                                        [PLACEHOLDER-293]=
                                                        [SINGLE_ID_AS_EXPR]
                                                            [ID]aa
                                                        [PLACEHOLDER-290];
                                                [SD_LIST]
                                                    [DECL_AS_SD]
                                                        [GLOBAL_DECL]
                                                            [PLACEHOLDER-261]global
                                                            [ID]aaa
                                                            [PLACEHOLDER-293]=
                                                            [SINGLE_ID_AS_EXPR]
                                                                [ID]aa
                                                            [PLACEHOLDER-290];
                                                    [SD_LIST]
                                                        [STATEMENT_AS_SD]
                                                            [IF_ELSE_STATEMENT]
                                                                [PLACEHOLDER-266]if
                                                                [BOOLEXPR]
                                                                    [ARRAY_REF_AS_EXPR]
                                                                        [ID]a
                                                                        [PLACEHOLDER-288][
                                                                        [ARRAY_REF_AS_EXPR]
                                                                            [ID]aa
                                                                            [PLACEHOLDER-288][
                                                                            [SINGLE_ID_AS_EXPR]
                                                                                [ID]aaa
                                                                            [PLACEHOLDER-289]]
                                                                        [PLACEHOLDER-289]]
                                                                    [PLACEHOLDER-282]==
                                                                    [ARRAY_REF_AS_EXPR]
                                                                        [ID]aaa
                                                                        [PLACEHOLDER-288][
                                                                        [TUPLE_REF_AS_EXPR]
                                                                            [ID]a
                                                                            [PLACEHOLDER-277].
                                                                            [INT]7
                                                                        [PLACEHOLDER-289]]
                                                                [PLACEHOLDER-267]then
                                                                [STATEMENT_LIST]
                                                                    [RETURN_STATEMENT]
                                                                        [PLACEHOLDER-273]return
                                                                        [ARRAY_REF_AS_EXPR]
                                                                            [ID]aaa
                                                                            [PLACEHOLDER-288][
                                                                            [TUPLE_REF_AS_EXPR]
                                                                                [ID]a
                                                                                [PLACEHOLDER-277].
                                                                                [INT]5
                                                                            [PLACEHOLDER-289]]
                                                                        [PLACEHOLDER-290];
                                                                    [EMPTY_NODE]
                                                                [EMPTY_NODE]
                                                                [ELSE_SENTENCE]
                                                                    [PLACEHOLDER-269]else
                                                                    [STATEMENT_LIST]
                                                                        [LHS_EXCHANGE_LHS_AS_STATEMENT]
                                                                            [LHS]
                                                                                [ARRAY_REF_AS_LHSITEM]
                                                                                    [ID]a
                                                                                    [PLACEHOLDER-288][
                                                                                    [ARRAY_REF_AS_EXPR]
                                                                                        [ID]a
                                                                                        [PLACEHOLDER-288][
                                                                                        [SINGLE_INT_AS_EXPR]
                                                                                            [INT]7
                                                                                        [PLACEHOLDER-289]]
                                                                                    [PLACEHOLDER-289]]
                                                                                [EMPTY_NODE]
                                                                            [PLACEHOLDER-294]<->
                                                                            [LHS]
                                                                                [TUPLE_REF_AS_LHSITEM]
                                                                                    [ID]a
                                                                                    [PLACEHOLDER-277].
                                                                                    [INT]3
                                                                                [EMPTY_NODE]
                                                                            [PLACEHOLDER-290];
                                                                        [STATEMENT_LIST]
                                                                            [RETURN_STATEMENT]
                                                                                [PLACEHOLDER-273]return
                                                                                [ARRAY_REF_AS_EXPR]
                                                                                    [ID]a
                                                                                    [PLACEHOLDER-288][
                                                                                    [ARRAY_REF_AS_EXPR]
                                                                                        [ID]a
                                                                                        [PLACEHOLDER-288][
                                                                                        [TUPLE_REF_AS_EXPR]
                                                                                            [ID]a
                                                                                            [PLACEHOLDER-277].
                                                                                            [INT]3
                                                                                        [PLACEHOLDER-289]]
                                                                                    [PLACEHOLDER-289]]
                                                                                [PLACEHOLDER-290];
                                                                            [EMPTY_NODE]
                                                                [PLACEHOLDER-263]end
                                                                [PLACEHOLDER-266]if
                                                        [EMPTY_NODE]
                                        [PLACEHOLDER-263]end
                                        [PLACEHOLDER-262]defun
                                [SDD_LIST]
                                    [STATEMENT_AS_SDD]
                                        [WHILE_STATEMENT]
                                            [PLACEHOLDER-264]while
                                            [BOOLEXPR]
                                                [SINGLE_ID_AS_EXPR]
                                                    [ID]a
                                                [PLACEHOLDER-283]!=
                                                [TUPLE_REF_AS_EXPR]
                                                    [ID]a
                                                    [PLACEHOLDER-277].
                                                    [INT]7
                                            [PLACEHOLDER-265]do
                                            [STATEMENT_LIST]
                                                [LHS_EXCHANGE_LHS_AS_STATEMENT]
                                                    [LHS]
                                                        [SINGLE_ID_AS_LHSITEM]
                                                            [ID]a
                                                        [EMPTY_NODE]
                                                    [PLACEHOLDER-294]<->
                                                    [LHS]
                                                        [SINGLE_ID_AS_LHSITEM]
                                                            [ID]a
                                                        [COMMA_LHSITEN_LIST]
                                                            [PLACEHOLDER-276],
                                                            [SINGLE_ID_AS_LHSITEM]
                                                                [ID]a
                                                            [EMPTY_NODE]
                                                    [PLACEHOLDER-290];
                                                [EMPTY_NODE]
                                            [PLACEHOLDER-263]end
                                            [PLACEHOLDER-264]while
                                    [EMPTY_NODE]
